index.js

```diff
@@ -0,0 +1,81 @@
+const D = require('date-fp');
+const S = require('sanctuary');
+const $ = require('sanctuary-def');
+
+const a = $.TypeVariable('a');
+const b = $.TypeVariable('b');
+
+const $TimeUnit = $.EnumType([
+  'milliseconds',
+  'seconds',
+  'minutes',
+  'hours',
+  'days']);
+
+const $TimeUnitFull = $.EnumType([
+  'milliseconds',
+  'seconds',
+  'minutes',
+  'hours',
+  'days',
+  'months',
+  'years']);
+
+const $TimeUnitGet = $.EnumType([
+  'milliseconds',
+  'seconds',
+  'minutes',
+  'hours',
+  'date',
+  'month',
+  'year']);
+
+const $Either = $.BinaryType(
+  'sanctuary/Either',
+  function(x) { return x != null && x['@@type'] === 'sanctuary/Either'; },
+  function(either) { return either.isLeft ? [either.value] : []; },
+  function(either) { return either.isRight ? [either.value] : []; }
+);
+
+const $InvalidDate = $.NullaryType(
+  'sanctuary-date-poc/Either',
+  function(x) {return !D.isValid(x)}
+);
+
+const env = $.env.concat([$.ValidDate, $InvalidDate, $.ValidNumber, $.Integer, $TimeUnit,
+  $TimeUnitFull, $TimeUnitGet, $Either]);
+const def = $.create(env);
+
+const add = (step, count, date) => {
+  const add = D.add(step, count, date);
+  return D.isValid(add) ? S.Right(add) : S.Left(add)
```

Comment:
This seems a bit odd. I'd expect the failure value to be `S.Left('Descriptive error message')`.